src/com/cft/contactmerge/contact/Address.java

```diff
@@ -77,6 +79,40 @@ public void setZip(String zip) {
      *************************************************** Sub-Methods ***************************************************
      *******************************************************************************************************************/
     public String toString() {
-        return getStreetAddress() + ", " + getApartment() + ", " + getCity() + ", " +
-                getState() + ", " + getCountry() + ", " + getZip(); }
+        StringBuilder fullAddress = new StringBuilder();
+        String[] addressParts = {getStreetAddress(), getApartment(), getCity(), getState(), getCountry(), getZip()};
+
+        for (String part: addressParts){
+            if (!part.isEmpty()){
+                fullAddress.append(part).append(", ");
+            }
+        }
+
+        if (!fullAddress.toString().isEmpty()){
+            fullAddress.delete(fullAddress.length() - 2, fullAddress.length());
+        }
+
+        return fullAddress.toString();
+    }
+
+    public AnswerType isAddressMatch(Address otherAddress){
```

Comment:
My current thoughts:
1. All address parts except apartment and zip are required. We won't add an address without these and so you shouldn't have to worry about getting and address without StreetAddress, City, State, or Country.
2. It is only a yes if all parts match. 
3. There are only three maybe cases I can think of that will result in a Address match of maybe : a) the StreetAddress match is a maybe and everything else is a yes; b) the StreetAddress is a yes or maybe and the Apartment is missing on one of the addresses being compared... everything else matches; c) the StreetAddress is a yes or maybe and the Zip is a maybe or missing... everything else matches.

Reply:
Thanks.